lib/public/components/common/dataExport/exportTriggerAndModal.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+import spinner from '../spinner.js';
+import { h } from '/js/src/index.js';
+
+/**
+ * Export form component, containing the fields to export, the export type and the export button
+ *
+ * @param {DataExportModel} exportModel export model
+ * @param {ModalHandler} modalHandler The modal handler, used to dismiss modal after export
+ *
+ * @return {Component} the form component
+ */
+const exportForm = (exportModel, modalHandler) => {
+    const exportTypes = ['JSON', 'CSV'];
+
+    const { selectedFields } = exportModel;
+    const { selectedExportType } = exportModel;
+    const fields = Object.keys(exportModel.dataExportConfiguration);
```

Comment:
Why not de-structuring the `dataExportConfiguration` at the same tie as the other two attributes?

Reply:
applied